Extensions/ITokenAcquisition.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Identity.Client;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace TodoListService.Extensions
+{
+    public interface ITokenAcquisition
+    {
+        /// <summary>
+        /// Add, to the MSAL.NET cache, the account of the user for which a bearer token was received when the Web API was called.
+        /// An On-behalf-of token is added to the cache, so that it can then be used to acquire another token on On-behalf-of the 
+        /// same user in order to call to downstream APIs.
+        /// </summary>
+        /// <param name="userAccessToken">Access token used to call this Web API</param>
+        /// <example>
+        /// From the configuration of the Authentication of the ASP.NET Core Web API: 
+        /// <code>JwtBearerOptions option;</code>
+        /// 
+        /// Subscribe to the token validated event:
+        /// <code>
+        /// options.Events = new JwtBearerEvents();
+        /// options.Events.OnTokenValidated = OnTokenValidated;
+        /// }
+        /// </code>
+        /// 
+        /// And then in the OnTokenValidated method, call <see cref="AddAccountToCache(JwtSecurityToken)"/>:
+        /// <code>
+        /// private async Task OnTokenValidated(TokenValidatedContext context)
+        /// {
+        ///  JwtSecurityToken accessToken = context.SecurityToken as JwtSecurityToken;
+        ///  _tokenAcquisition.AddAccountToCache(accessToken);
+        /// }
+        /// </code>
+        /// </example>
+        void AddAccountToCacheFromJwt(JwtSecurityToken jwtToken, IEnumerable<string> scopes);
+
+        Task AddAccountToCacheFromAuthorizationCode(AuthorizationCodeReceivedContext context, IEnumerable<string> scopes);
```

Comment:
comment missing?

Reply:
I added in the comment, but I'm not sure if it's good enough.
@jmprieur Would you be able to take a look?